webv2/index/index.go

```diff
@@ -0,0 +1,209 @@
+package index
+
+import (
+	"fmt"
+
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+	"github.com/cloudspannerecosystem/harbourbridge/webv2/session"
+	utilities "github.com/cloudspannerecosystem/harbourbridge/webv2/utilities"
+)
+
+// IndexSuggestion adds redundant index and interleved index  suggestion in schema conversion process for database.
+func IndexSuggestion() {
+
+	sessionState := session.GetSessionState()
+
+	for _, spannerTable := range sessionState.Conv.SpSchema {
+		CheckIndexSuggestion(spannerTable.Indexes, spannerTable)
+	}
+}
+
+//Helpers method for checking Index Suggestion.
+func CheckIndexSuggestion(index []ddl.CreateIndex, spannerTable ddl.CreateTable) {
+
+	redundantIndex(index, spannerTable)
+	interleaveIndex(index, spannerTable)
+}
+
+// redundantIndex check for redundant Index.
+// If present adds IndexRedandant as an issue in Issues.
+func redundantIndex(index []ddl.CreateIndex, spannerTable ddl.CreateTable) {
+
+	for i := 0; i < len(index); i++ {
+
+		keys := index[i].Keys
+		var primaryKeyFirstColumn string
+		pks := spannerTable.Pks
+
+		for i := range pks {
+			if pks[i].Order == 1 {
+				primaryKeyFirstColumn = pks[i].Col
+			}
+		}
+
+		indexFirstColumn := index[i].Keys[0].Col
```

Comment:
This is taking the assumption that session file will have indexes in sorted order which may not be true if at some point we give user the option to edit the ordering